specs/spec_reader.go

```diff
@@ -173,3 +182,34 @@ func NewSpecReader(paths []string) (*SpecReader, error) {
 
 	return reader, nil
 }
+
+// strip yaml comments from the given yaml document by converting to JSON and back :)
+func stripYamlComments(b []byte) ([]byte, error) {
+	const openPlaceholder = "$$$OPEN$$$"
+	const closePlaceholder = "$$$CLOSE$$$"
+
+	// return an error if the yaml already contains our temporary placeholder for env variables by some unlucky coincidence
+	if bytes.Contains(b, []byte(openPlaceholder)) || bytes.Contains(b, []byte(closePlaceholder)) {
+		return nil, fmt.Errorf("%s and %s are reserved words in CloudQuery config", openPlaceholder, closePlaceholder)
+	}
+
+	// replace placeholder variables with valid yaml, otherwise it cannot be parsed
+	// in some cases. Short of writing our own yaml parser to remove comments,
+	// this seems like the best we can do.
+	b = envRegex.ReplaceAllFunc(b, func(match []byte) []byte {
+		content := envRegex.FindSubmatch(match)[1]
+		return []byte(openPlaceholder + string(content) + closePlaceholder)
+	})
+	j, err := yaml.YAMLToJSON(b)
```

Comment:
already done in `SpecUnmarshalYamlStrict`, maybe we could do yaml-> json and then work with json only?

Reply:
We would then be breaking some functionality that we know some users are relying on (and requested several times in the past when we didn't have it, as I was informed by @bbernays): the ability to store actual yaml inside environment variables and having it expand. I am of the opinion that doing this is a terrible idea, but for now I just want to fix the thing we can all agree needs to be fixed, which is that environment variables in comments should be ignored.